### PR TITLE
Mm 41869 learn more about start trial modal

### PR DESCRIPTION
**Pull request by pvev**
#### Summary
This PR adds the changes to the start menu call to action to review the benefits that a client gets by starting the trial. By clicking the "Learn more" link/button from the CTA section, a modal is presented, exposing the benefits of starting the trial and calling to action to actually start it.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-41869

#### Screenshots

https://user-images.githubusercontent.com/10082627/155223988-d895cd41-f879-406e-adb7-1752e98c9439.mp4

https://user-images.githubusercontent.com/10082627/155224030-609cd873-10fa-4053-86de-a6c3ce56edc0.mp4

https://user-images.githubusercontent.com/10082627/155224038-8caf4ef3-8fb0-427b-9d2b-a5d2016d87bc.mp4

Changes in the Onboarding Task List:

https://user-images.githubusercontent.com/10082627/155842041-2aeebe9d-3c8a-429d-a4d9-b3f31ebfcf5d.mp4

https://user-images.githubusercontent.com/10082627/155842044-c39ce8ad-bf5b-4dba-8785-f015b5039888.mp4

#### Release Note
```release-note
The start menu start-trial CTA presents a modal exposing the benefits the client gets by starting the trial, encouraging admins to request a trial license and engage them to the product, this is expected to increase product adoption
```

### CODE REVIEW

File: components/common/carousel/carousel.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ type Props = {
     onNextSlideClick?: (slideIndex: number) => void;
     onPrevSlideClick?: (slideIndex: number) => void;
 }
-const Carousel: React.FC<Props> = ({
```
Comment (pablovelezvidal) [inline components/common/carousel/carousel.tsx:15]: this is just part of the campaign to remove the type React.FC from components that does not expect children

File: components/common/carousel/carousel.tsx [hunk 2]
```diff
@@ -77,7 +77,7 @@ const Carousel: React.FC<Props> = ({
             {dataSlides.map((obj: any, index: number) => {
                 return (
                     <div
-                        key={`${index}`}
+                        key={`${index.toString()}`}
```
Comment (pablovelezvidal) [inline components/common/carousel/carousel.tsx:80]: lintern fix

File: components/common/carousel/carousel.tsx [hunk 3]
```diff
@@ -89,7 +89,7 @@ const Carousel: React.FC<Props> = ({
                 <div className='container-dots'>
                     {dataSlides.map((item, index) => (
                         <div
-                            key={index}
+                            key={index.toString()}
```
Comment (pablovelezvidal) [inline components/common/carousel/carousel.tsx:92]: lintern fix

File: components/generic_modal.tsx [hunk 4]
```diff
@@ -139,22 +139,20 @@ export default class GenericModal extends React.PureComponent<Props, State> {
                 <Modal.Header
                     closeButton={true}
                 />
```
Comment (pablovelezvidal) [inline components/generic_modal.tsx:141]: removing form tag. I don't see any use case for keeping it here. @devinbinnie do you remember on top of your mind a specific reason why the generic modal content is surrounded by a <form> tag?

File: components/trial_benefits_modal/trial_benefits_modal.tsx [hunk 5]
```diff
@@ -181,20 +211,75 @@ const TrialBenefitsModal = (
...
 
-    return (
-        <GenericModal
-            className='TrialBenefitsModal'
-            show={show}
-            id='trialBenefitsModal'
-            onExited={handleOnClose}
-        >
+    const trialJustStartedScreen = (declaration: Omit<TrialBenefitsModalStepProps, 'pageURL' | 'buttonLabel'>) => {
+        const {id, title, description, svgWrapperClassName, svgElement, bottomLeftMessage} = declaration;
```
Comment (julmondragon) [inline components/trial_benefits_modal/trial_benefits_modal.tsx:215]: What about:
```
const trialJustStartedScreen = ({id, title, description, svgWrapperClassName, svgElement, bottomLeftMessage}:  Omit<TrialBenefitsModalStepProps, 'pageURL' | 'buttonLabel'>) => (
```

Comment (julmondragon) [reply components/trial_benefits_modal/trial_benefits_modal.tsx:215]: In a recent change (https://github.com/mattermost/mattermost-webapp/pull/9803) `buttonLabel` prop was added to `TrialBenefitsModalStep`, which is actually only required when `pageURL` is provided.
So maybe should set `buttonLabel` also as optional, then wont need the Omit here.

File: components/learn_more_trial_modal/learn_more_trial_modal.tsx [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+            svgWrapperClassName: 'personBoxSvg',
+            svgElement: (
+                <PushNotificationsSvg
+                    width={400}
+                    height={200}
+                />
+            ),
+            buttonLabel: formatMessage({id: 'benefits_trial.modal.learnMore', defaultMessage: 'Learn More'}),
+        },
+    ], []);
```
Comment (julmondragon) [inline components/learn_more_trial_modal/learn_more_trial_modal.tsx:90]: In a recent change (https://github.com/mattermost/mattermost-webapp/pull/9803) a new step was added for Playbooks. Do we need to add it here as well?

**Comment by stevemudie** [comment]
This is looking good @pablovelezvidal but I noticed a couple of issues along the way.

When I started the trial on the System Console part of the modal it loaded a second modal on top of the original. Is this expected?
Screen Shot 2022-02-23 at 4 24 31 PM

Also, once the trial was i play and I was exploring the benefits the one to check out Playbooks is a dead end even though you can get to Playbooks by way of the menu.

**Review by AGMETEOR — APPROVED** [review]

**Review by AGMETEOR — APPROVED** [review]
Code LGTM

File: i18n/en.json [hunk 7]
```diff
@@ -3604,6 +3604,7 @@
   "last_users_message.others": "{numOthers} others ",
   "last_users_message.removed_from_channel.type": "were **removed from the channel**.",
   "last_users_message.removed_from_team.type": "were **removed from the team**.",
+  "learn_more_trial_modal.pretitle": "With Enterprise, you can....",
```
Comment (julmondragon) [inline i18n/en.json:3607]: Four dots.

**Review by julmondragon — APPROVED** [review]
LGTM!

File: components/onboarding_tasklist/onboarding_tasklist_completed.tsx [hunk 8]
```diff
@@ -122,26 +142,16 @@ const Completed = (props: Props): JSX.Element => {
...
-                            target='_blank'
-                            rel='noopener noreferrer'
-                            href={'https://docs.mattermost.com/'}
-                        >
-                            <FormattedMessage
-                                id='onboardingTask.checklist.documentation_link'
-                                defaultMessage='See our documentation.'
-                            />
-                        </a>
-                    </p>
```
Comment (marianunez) [inline components/onboarding_tasklist/onboarding_tasklist_completed.tsx:140]: Is this help link removal intended?

File: components/onboarding_tasklist/onboarding_tasklist_completed.tsx [hunk 9]
```diff
@@ -150,13 +160,12 @@ const Completed = (props: Props): JSX.Element => {
                                     id='onboardingTask.checklist.start_enterprise_now'
                                     defaultMessage='Start your free Enterprise trial now!'
```
Comment (marianunez) [inline components/onboarding_tasklist/onboarding_tasklist_completed.tsx:161]: I believe the latest message for is `Learn More...` can you double check the Figma for this?

Comment (marianunez) [reply components/onboarding_tasklist/onboarding_tasklist_completed.tsx:161]: Nvm, I see this is for the completed step now.

File: components/generic_modal.scss [hunk 10]
```diff
@@ -114,7 +114,7 @@
 .GenericModal__body {
     position: relative;
     width: 100%;
-    padding: 0 24px;
+    padding: 0 24px 24px 24px;
```
[the remark this answers is not in the document]
Comment (pablovelezvidal) [reply components/generic_modal.scss:117]: True, good catch! could potentially cause some changes on how content look, so it makes sense to do what you suggest and add it in a more higher level file. Thanks!

File: components/onboarding_tasklist/onboarding_tasklist_completed.tsx (hunk 8 above)
Comment (pablovelezvidal) [reply components/onboarding_tasklist/onboarding_tasklist_completed.tsx:140]: yes, according to the design this gets removed.
Captura de pantalla 2022-03-03 a las 7 50 17

**Comment by pablovelezvidal** [comment]
@stevemudie the issues you saw during your initial review are solved with this PR that is in pending review state https://github.com/mattermost/enterprise/pull/1150 I will let you know when it gets merge and we can get back to this one. Thanks!

**Review by stevemudie — APPROVED** [review]
Everything is working as expected. You should be all set @pablovelezvidal!

**Review by AGMETEOR — APPROVED** [review]
LGTM

**Comment by pablovelezvidal** [comment]
@annelieseklein gentle ping on this one 🙏

File: components/generic_modal.tsx [hunk 11]
```diff
@@ -139,22 +139,20 @@ export default class GenericModal extends React.PureComponent<Props, State> {
...
-                <form>
```
Comment (pablovelezvidal) [inline components/generic_modal.tsx:142]: @julmondragon this is the <form> tag belonging to the generic modal that is causing you issues when using your new input widget. For what I can see it is safe to remove, actually, I am removing it in this PR and besides all tests passing, I checked some places where the generic modal is used and seems to continue working fine, but I want to confirm if it was just a html-tag leftover or was set here intentionally. Wondering if @devinbinnie remembers if there was a reason for it?

Comment (devinbinnie) [reply components/generic_modal.tsx:142]: I don't explicitly remember if there was, but my guess was to follow semantic HTML for accessibility reasons. Just spitballing here though.

Comment (marianunez) [reply components/generic_modal.tsx:142]: @pablovelezvidal can you list the modals that use this `GenericModal` so @stevemudie can have at pass at them in this PR and make sure they are working as expected still?

Comment (pablovelezvidal) [reply components/generic_modal.tsx:142]: Thanks @marianunez, Sure:

1. Confirm license removal modal
2. Enterprise Edition information agreement modal (licensing pages in starter mode have some links to show a modal witht the EE agreement info)
3. Upload license modal
4. No internet connection modal (announcement bar shows info about renewing the license, when you click and somehow there is no internet access that modal is presented)
5. Collapse/Reply Threads Beta modal (I think this is in beta state and not sure how to get it shown)
6. Collapse/Reply Threads modal
7. Custom Status Modal
8. Delete Category Modal
9. DnD Custom timepicker modal
10. Edit category modal
11. Product Notices modal
12. Remove next steps modal
13. Trial Benefits modal

@stevemudie let me know if you have any question or  face any issues trying to validate this.

**Comment by annelieseklein** [comment]
Screen Shot 2022-03-10 at 10 52 08 AM
@pvev the active state of this button is incorrect and should have no underline for the text.

Here is the file for the button component: https://www.figma.com/file/2uYWxjnMJ9IQDOba9b9bfV/Components---Buttons?node-id=1%3A184

**Review by annelieseklein — APPROVED** [review]

File: components/generic_modal.tsx (hunk 11 above)
Comment (pvev) [reply components/generic_modal.tsx:142]: ok, we re-reviewed all these modals and everything seems to be working fine. Thanks @stevemudie for the help.

**Pull request merged**